stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -2473,6 +2473,11 @@ collections = [
 /// keys, followed by the values.
 //
 // See the docs at the top of the file for more details on this type
+//
+// NOTE: The precise layout of this type is relied on in the runtime
+// to provide a statically allocated empty singleton. 
```

Comment:
In the interest of incremental improvement, are you fine with moving forward with this PR as-is for now? The Dictionary/Array layouts at this level in the system are basically finalized, so the only thing that could break these are someone implementing a new class layout algorithm.